src/main/java/com/google/sps/data/User.java

```diff
@@ -55,33 +57,47 @@
   private Entity entity;
 
   /**
-   * This creates a standard universal user instance regardless of login status.
-   * This method will be changed to instantiate distinct user object.
+   * Verifies user and creates a User instance for accessing and adding user data.
+   * throws SecurityException on failure, import from java.lang.SecurityException.
+   * @Param idTokenString is the Google-user ID Token provided buy user-auth.js/getIdToken.
+   * Token should be passed as URL Fetch argument from front end.
    */
-  public User() {
-    String testId = TEST_ID;
-    Key userKey = KeyFactory.createKey("User", testId);
+  public User(String idTokenString) throws SecurityException {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier
+    .Builder(UrlFetchTransport.getDefaultInstance(), new JacksonFactory())
+    .setAudience(Collections.singletonList(CLIENT_ID))
+    .build();
+
+    GoogleIdToken idToken;
+    try {
+      idToken = verifier.verify(idTokenString);
+    } catch(IOException | GeneralSecurityException e) {
+      throw new SecurityException("Failed to verify Google token", e);
+    }
+
+    payload = idToken.getPayload();
+
+    Key userKey = KeyFactory.createKey("User", getId());
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     try {
       entity = datastore.get(userKey);
```

Comment:
Are you doing anything with the entity here, or is this basically a check to see if the user already exists? If you wouldn't mind adding a comment here that'd be great.